src/views/ModelPlan/TaskList/ParticipantsAndProviders/_component/ProviderAndSupplierModal/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import ReactModal from 'react-modal';
+import {
+  Button,
+  Grid,
+  GridContainer,
+  Icon,
+  SideNav as TrussSideNav
+  // Table as UswdsTable
+} from '@trussworks/react-uswds';
+
+import useCheckResponsiveScreen from 'hooks/useCheckMobile';
+import {
+  ExistingProviderSupplierTypes,
+  existingProviderSupplierTypesNames
+} from 'i18n/en-US/modelPlan/participantsAndProviders';
+
+import ProviderAndSupplierTable from '../ProviderAndSupplierTable';
+
+import './index.scss';
+
+type ProviderAndSupplierModalProps = {
+  isOpen: boolean;
+  closeModal: () => void;
+};
+
+const ProviderAndSupplierModal = ({
+  isOpen,
+  closeModal
+}: ProviderAndSupplierModalProps) => {
+  const { t: modalT } = useTranslation('participantsAndProvidersMisc');
+  const isMobile = useCheckResponsiveScreen('tablet');
+
+  const [activeType, setActiveType] = useState<ExistingProviderSupplierTypes>(
+    ExistingProviderSupplierTypes.PROVIDER_TYPES_INSTITUTIONAL
+  );
+
+  const providerAndSupplierTypes: Record<
+    ExistingProviderSupplierTypes,
+    string
+  > = modalT('modal.existingProviderSupplierTypesNames', {
+    returnObjects: true
+  });
+
+  const sidenavItems = Object.entries(providerAndSupplierTypes).map(
+    ([key, value]) => {
+      return (
+        <Button
+          type="button"
+          unstyled
+          className={`sidenav-button line-height-sans-3 width-full ${
+            activeType === key ? 'current' : ''
+          }`}
+          onClick={() => setActiveType(key as ExistingProviderSupplierTypes)}
+        >
+          {value}
+        </Button>
+      );
+    }
+  );
```

Comment:
You can use `geyKeys` here to preserve the type and not need the type assertion `key as ExistingProviderSupplierTypes` I believe.